packages/eslint-plugin/tests/rules/no-unsafe-unary-minus.test.ts

```diff
@@ -0,0 +1,47 @@
+import { RuleTester } from '@typescript-eslint/rule-tester';
+
+import rule from '../../src/rules/no-unsafe-unary-minus';
+import { getFixturesRootDir } from '../RuleTester';
+
+const rootDir = getFixturesRootDir();
+const ruleTester = new RuleTester({
+  parserOptions: {
+    ecmaVersion: 2015,
+    tsconfigRootDir: rootDir,
+    project: './tsconfig.json',
+  },
+  parser: '@typescript-eslint/parser',
+});
+
+ruleTester.run('no-unsafe-unary-minus', rule, {
+  valid: [
+    '+42;',
+    '-42;',
+    '-42n;',
+    '(a: number) => -a;',
+    '(a: bigint) => -a;',
+    '(a: number | bigint) => -a;',
+    '(a: any) => -a;',
+    '(a: 1 | 2) => -a;',
+    '(a: string) => +a;',
```

Comment:
[Functionality] ...huh. I at first thought this was a bug, and that `+` shouldn't be allowed. But the rule is just for `unary-minus`. Maybe this is evidence that the rule should be a more broad `no-unsafe-unary`, for `+` on non-numeric values too? Thoughts?

(if so we should probably discuss in the issue)

Reply:
`+x` is a common synonym for `Number(x)`, so this will surely make some unhappy.

Reply:
Yeah, we could check `+` too, but in my opinion there's no need: any person using this rule would probably already be using ESLint's [no-implicit-coercion](https://eslint.org/docs/latest/rules/no-implicit-coercion) rule, which would already catch all instances of unary `+`. I could remove this particular test case if you want, though, if it's confusing.

Reply:
Right, that makes sense. I think it's fine as-is. @Josh-Cena if you want to push either direction I'll go with that 😄

Reply:
Yes, let's keep it as-is. I don't think it's worth removing either—always good to delineate the boundaries of a rule so future behavior changes are obvious.